.github/workflows/ci.yml

```diff
@@ -85,27 +86,35 @@ jobs:
       - name: Checkout
         uses: actions/checkout@master
 
+      - name: "Install PHP with extensions"
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: xdebug
+          extension-csv: dom, json, libxml, xml, xmlwriter
+          ini-values-csv: assert.exception=1, zend.assertions=1
+
       - name: "Cache dependencies installed with composer"
         uses: actions/cache@v1
         with:
           path: ~/.composer/cache
-          key: ${{ matrix.php-binary }}-composer-${{ matrix.dependencies }}-${{ hashFiles('**/composer.json') }}
+          key: php${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-${{ hashFiles('**/composer.json') }}
           restore-keys: |
-            ${{ matrix.php-binary }}-composer-${{ matrix.dependencies }}-
+            php${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-
 
       - name: Install lowest dependencies with composer
         if: matrix.dependencies == 'lowest'
-        run: ${{ matrix.php-binary }} ./tools/composer update --no-ansi --no-interaction --no-progress --no-suggest --prefer-lowest
+        run: ./tools/composer update --no-ansi --no-interaction --no-progress --no-suggest --prefer-lowest
 
       - name: Install highest dependencies with composer
         if: matrix.dependencies == 'highest'
-        run: ${{ matrix.php-binary }} ./tools/composer update --no-ansi --no-interaction --no-progress --no-suggest
+        run: ./tools/composer update --no-ansi --no-interaction --no-progress --no-suggest
 
       - name: Run sanity check
-        run: PHP_BINARY=${{ matrix.php-binary }} bash ./build/scripts/sanity-check
+        run: bash ./build/scripts/sanity-check
 
       - name: Run tests with phpunit
-        run: ${{ matrix.php-binary }} -d assert.exception=1 -d zend.assertions=1 ./phpunit --coverage-clover=coverage.xml
```

Comment:
I've set these values via the `ini-values-csv` section above, does that make sense?